src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/WorkingDaysService.kt

```diff
@@ -7,29 +7,43 @@ import java.time.LocalDate
 @Service
 class WorkingDaysService(private val bankHolidayService: BankHolidayService) {
 
-  fun addWorkingDays(date: LocalDate, workingDays: Int): LocalDate {
-    val bankHolidays = bankHolidayService.getBankHolidaysForEnglandAndWales()
-    var adjustedDate = date
-    for (i in 0 until workingDays) {
-      adjustedDate = getNextWorkingDay(bankHolidays, adjustedDate)
-    }
-    return adjustedDate
+  fun workingDaysAfter(
+    date: LocalDate,
+  ): Sequence<LocalDate> {
+    return generateSequence(date) { it.nextWorkingDay() }.drop(1)
   }
 
-  fun isWeekend(date: LocalDate): Boolean {
-    val weekend = listOf(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY)
+  fun workingDaysBefore(
+    date: LocalDate,
+  ): Sequence<LocalDate> {
+    return generateSequence(date) { it.previousWorkingDay() }.drop(1)
+  }
+
+  fun isWeekend(date: LocalDate, weekend: List<DayOfWeek> = listOf(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY)): Boolean {
     return date.dayOfWeek in weekend
   }
 
-  fun isNonWorkingDay(bankHolidays: List<LocalDate>, date: LocalDate): Boolean {
-    return isWeekend(date) || bankHolidays.contains(date)
+  fun isNonWorkingDay(date: LocalDate, weekend: List<DayOfWeek> = listOf(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY)): Boolean {
+    return isWeekend(date, weekend) || getBankHolidays().contains(date)
   }
 
-  fun getNextWorkingDay(bankHolidays: List<LocalDate>, date: LocalDate): LocalDate {
-    var adjustedDate = date.plusDays(1)
-    while (isNonWorkingDay(bankHolidays, adjustedDate)) {
+  fun LocalDate.nextWorkingDay(): LocalDate {
+    var adjustedDate = this.plusDays(1)
```

Comment:
are these used? if so could rewrite to use the sequence, if not could remove?

Reply:
oh, might be better to implement this as a filter on the sequence - the sequence does the iteration/ generation of new values

Reply:
I get what you mean - let me fix that.